src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs

```diff
@@ -35,9 +35,10 @@ public ConfigurationReader(IConfiguration configuration, DependencyContext depen
 
         // Generally the initial call should use IConfiguration rather than IConfigurationSection, otherwise
         // IConfiguration parameters in the target methods will not be populated.
-        ConfigurationReader(IConfigurationSection configSection, DependencyContext dependencyContext)
+        public ConfigurationReader(IConfigurationSection configSection, DependencyContext dependencyContext)
         {
-            _section = configSection ?? throw new ArgumentNullException(nameof(configSection));
+            _configuration = configSection ?? throw new ArgumentNullException(nameof(configSection));
```

Comment:
It was probably an oversight to leave `public` off this, but the comment mentions that `IConfiguration` parameters won't be populated in target methods, and populating `_configuration` isn't likely to help. The only place this matters that I'm aware of (so far) is in the SQL sink where the full config is used to look for named connection strings (which would be a config section external to the Serilog section, whatever name it might use).

I'd have to study this and think about the issue more but offhand I can't think of any other problems fixing it this way. I'd just leave `_configuration` null (without looking at the code, I assume the package would treat a null as not having an `IConfiguration` available to populate in the target method).